src/lib/Components/ContextMenu.tsx

```diff
@@ -18,15 +18,29 @@ import { getSelectedLocations } from "../Functions/getSelectedLocations";
 import { useReactGridState } from "./StateProvider";
 
 export const ContextMenu: React.FC = () => {
+  const targetRef = React.useRef<HTMLDivElement>(null);
   const state = useReactGridState();
-
-  if (
-    state.contextMenuPosition.top === -1 &&
-    state.contextMenuPosition.left === -1
-  )
-    return null;
-
   const { contextMenuPosition, selectedIds, selectionMode } = state;
+  const clickX = contextMenuPosition.left;
+  const clickY = contextMenuPosition.top;
+  if (clickY === -1 && clickX === -1) return null;
+  const screenW = window.innerWidth;
+  const screenH = window.innerHeight;
+  const menuW = targetRef?.current?.offsetWidth || 194;
+  const menuH = targetRef?.current?.offsetHeight || 150;
```

Comment:
Where did you get the default values from?
Also: this will at least once resolve to the default values as the targetRef is null on first render.